src/context/src/renderer.rs

```diff
@@ -0,0 +1,407 @@
+            }
+            #[cfg(windows)]
+            VideoContext::Direct3D {  } => {
+                // stub
+                None
+            },
+            VideoContext::Null => None,
+        }
+    }
+    /// Delete a `Camera`.
+    pub fn delete_camera(&mut self, name: &str) {
+        match self.video_context {
+            VideoContext::OpenGL { ref mut frame, .. } => {
+                let name: String = name.into();
+                frame.cameras.remove(&name);
+            }
+            #[cfg(windows)]
+            VideoContext::Direct3D {  } => {
+                // stub
+            },
+            VideoContext::Null => (),
+        }
+    }
+
+    /// Create a `Fragment` from vertex data, ka texture, kd texture, and transform matrix.
+    pub fn create_fragment(&mut self, data: &Vec<VertexPosNormal>, ka: Texture, kd: Texture, transform: [[f32; 4]; 4]) -> Option<Fragment> {
+        match self.video_context {
+            VideoContext::OpenGL {
+                ref mut factory,
+                ..
+            } => {
+                let (buffer, slice) = factory.create_vertex_buffer_with_slice(&data, ());
+
+                let ka = match ka.texture_impl {
+                    TextureImpl::OpenGL { texture } => texture,
+                    #[cfg(windows)]
+                    TextureImpl::Direct3D {  } => return None,
+                    TextureImpl::Null => return None,
+                };
+
+                let kd = match kd.texture_impl {
+                    TextureImpl::OpenGL { texture } => texture,
+                    #[cfg(windows)]
+                    TextureImpl::Direct3D {  } => return None,
+                    TextureImpl::Null => return None,
+                };
+
+                let fragment = amethyst_renderer::Fragment {
+                    transform: transform,
+                    buffer: buffer,
+                    slice: slice,
+                    ka: ka,
+                    kd: kd,
+                };
+                let fragment_impl = FragmentImpl::OpenGL {
+                    fragment: fragment,
+                };
+                Some(Fragment {
+                    fragment_impl: fragment_impl,
+                })
+            },
+            #[cfg(windows)]
+            VideoContext::Direct3D {  } => {
+                // stub
+                None
+            },
+            VideoContext::Null => None,
+        }
+    }
+
+    /// Create a constant solid color `Texture`.
+    pub fn create_constant_texture(&self, color: [f32; 4]) -> Texture {
+        let texture = amethyst_renderer::Texture::Constant(color);
+        let texture_impl = TextureImpl::OpenGL {
+            texture: texture,
+        };
+        Texture {
+            texture_impl: texture_impl,
+        }
+    }
+
+    /// Create a `Texture` from pixel data.
+    pub fn create_texture(&mut self, kind: Kind, data: &[&[<<ColorFormat as Formatted>::Surface as SurfaceTyped>::DataType]]) -> Option<Texture> {
```

Comment:
so, just to be clear, this only supports a single format (`ColorFormat`), right?